source/extensions/clusters/dynamic_forward_proxy/cluster.cc

```diff
@@ -195,8 +188,14 @@ ClusterFactory::createClusterWithConfig(
     Stats::ScopePtr&& stats_scope) {
   Extensions::Common::DynamicForwardProxy::DnsCacheManagerFactoryImpl cache_manager_factory(
       context.singletonManager(), context.dispatcher(), context.tls(), context.stats());
+  envoy::config::cluster::v3::Cluster cluster_config = cluster;
+  if (!cluster_config.has_upstream_http_protocol_options()) {
+    cluster_config.mutable_upstream_http_protocol_options()->set_auto_sni(true);
+    cluster_config.mutable_upstream_http_protocol_options()->set_auto_san_validation(true);
```

Comment:
It would be nice to fail here if someone actually explicitly set these to false, but as they aren't WKT I don't think we can determine that, so shrug I guess? Maybe update the docs for those fields to say they are force set to true when using the dynamic forward proxy cluster?

Reply:
We can determine if there was explicit `{}` for upstream_http_protocol_options but not setting them true.